### PR TITLE
Status effects indicators battle

### PR DESCRIPTION
**Pull request by tkz00**
## Motivation

Status effects affecting the units must be communicated to the user.

## Summary of changes

When a `TagReceived` or `ModifierReceived` action is read from the battle history, the respective status icon to that condition is show above the affected unit's health bar (for example, a boot with a downwards arrow for a reduction of speed). This behavior is achieved through a new component in the BattleUnit prefab, `StatusIndicators`, which keeps track of current effects on the unity via dictionaries, so the same icon doesn't appear multiple times, and spawns and destroys the icons accordingly.

## How has this been tested?

Let a battle with units that apply tags or modifiers play out, when those effects are applied and removed, verify the corresponding icons appear and disappear from above the unit's health bar.

This is hard to do in the normal state of the PR since steps go by quickly, but you modify the code to pause the games execution with `Debug.Break()` and print the current step on the client, with this you can see the first modifier/tag icon that appears and the step it was applied on, keep in mind `Debug.Break()` isn't precise and will probably pause the execution one step after the one that applied the status effects, you can also print the name of the modifier/tag applied in the `ApplyStatus()` method of `BattleUnit.cs` for better clarity. With this information you can then check the history printed by the backend to see if the corresponding status was applied to the correct unit in the respective step.

To check which icons correspond with which modifier/tag check `Assets/ScroptableObjects/StatusEffects`

### CODE REVIEW

File: client/Assets/Scripts/Battle/BattleManager.cs [hunk 1]
```diff
@@ -84,19 +87,56 @@ private IEnumerator PlayOutSteps(RepeatedField<Protobuf.Messages.Step> steps)
 	{
 		foreach (var step in steps)
 		{
-			yield return new WaitForSeconds(.4f);
+			Debug.Log($"step: {step.StepNumber}");
```
Comment (ncontinanza) [inline client/Assets/Scripts/Battle/BattleManager.cs:90]: Do we need this comment? It's making the log huge on the client

**Review by ncontinanza — CHANGES_REQUESTED** [review]
Working great! Just left a comment about a log

File: client/Assets/Scripts/Battle/BattleManager.cs (hunk 1 above)
Comment (tkz00) [reply client/Assets/Scripts/Battle/BattleManager.cs]: Sorry, my bad, removed the line!

**Review by ncontinanza — APPROVED** [review]
Nice!

**Comment by tkz00** [comment]
> please use a different icon for Defense and Damage Reduction

Fixed!

**Review by ncontinanza — DISMISSED** [review]

File: client/Assets/Scripts/Battle/BattleManager.cs [hunk 2]
```diff
@@ -116,6 +119,62 @@ private IEnumerator PlayOutSteps(RepeatedField<Protobuf.Messages.Step> steps)
...
+							{
+								prefix = action.ModifierReceived.StatAffected.Amount > 0 ? "higher_" : "lower_";
+							}
+							else if (action.ModifierReceived.Operation == "Multiply")
+							{
+								prefix = action.ModifierReceived.StatAffected.Amount > 1 ? "higher_" : "lower_";
+							}
+							if (!statuses.Any(status => status.name.ToLower() == prefix + action.ModifierReceived.StatAffected.Stat.ToString().ToLower()))
+							{
+								Debug.LogError($"status not found on client: {prefix + action.ModifierReceived.StatAffected.Stat.ToString().ToLower()}");
```
Comment (lotuuu) [inline client/Assets/Scripts/Battle/BattleManager.cs:136]: I think logging these as errors is a bit too harsh. We're expecting to get a lot of tags that are more "internal use"-ish and won't receive animations/feedback. We can keep this log but maybe as a warning or even just info. Same for tags.

**Review by lotuuu — CHANGES_REQUESTED** [review]
Let's change the Paralyze icon since the ❄️ would belong to freeze! You can check possible status effects here: https://docs.google.com/spreadsheets/d/1qLoq8PrQpByb_TaQaxlm8lgQkd1mQSSW4dTxoc7M7fM/edit#gid=666137060

File: client/Assets/Scripts/Battle/BattleManager.cs (hunk 2 above)
Comment (tkz00) [reply client/Assets/Scripts/Battle/BattleManager.cs]: You're right! Changed

**Comment by tkz00** [comment]
> Let's change the Paralyze icon since the ❄️ would belong to freeze! You can check possible status effects here: https://docs.google.com/spreadsheets/d/1qLoq8PrQpByb_TaQaxlm8lgQkd1mQSSW4dTxoc7M7fM/edit#gid=666137060

Changed to a little guy being electrocuted!

**Review by ncontinanza — DISMISSED** [review]
LGTM!

**Review by ncontinanza — APPROVED** [review]

**Review by lotuuu — APPROVED** [review]

**Review by martinacantaro — APPROVED** [review]

**Pull request merged**